### PR TITLE
Highlightfix

### LINKED ISSUE #55: Hard to follow muted people's conversation
When a user gets muted in chat, his message and name get grayed out. 
If you click on their name after that, it is (too) hard to follow their conversaiton; colors should be adjusted.

muted: https://i.imgur.com/FAHg7Dh.png
muted and "highlighted": https://i.imgur.com/2YuXFfN.png

### PR DESCRIPTION
**Pull request by PinkishCow**
Ignore the commit autism, this doesn't work in the backlog, fuck knows why. 

Fixes #55 again.

### CODE REVIEW

**Comment by kenany** [comment]
No idea what's going on here

**Comment by PinkishCow** [comment]
After a total of 4 and a quarter hours, neither do I, it needs to be in 2 wraps of jquery, i found that from a page 2 google result, and it all looked like this at one point http://puu.sh/qnYFE/39b1daf73f.png and http://puu.sh/qnVMt/6c27813da3.png All i know is it works 100% with both capitalised and lowercase names. But it doesn't work in the backlog. just because
 it feels like

**Comment by PinkishCow** [comment]
Also gui.toggleUserFocus just didn't feel like having any of it so i had to copy the code out of it and do it manually once i managed to get the names that it needed to focus

File: betterdgg/modules/mentions.js [hunk 1]
```diff
@@ -7,9 +7,12 @@
                 gui.lines.on('mousedown', 'div.user-msg a.user', function() {
                     if (bdgg.settings.get('bdgg_highlight_selected_mentions')) {
                         var username = $(this).text();
-                        gui.lines.find(':contains(' + username + ')').addClass('focused');
+                        var users = [$(this).parents().parents().find(".chat-user:contains(" + username + ")")];
```
Comment (kenany) [inline betterdgg/modules/mentions.js:10]: Can this just be?:

``` javascript
var users = gui.lines.find(".chat-user:contains(" + username + ")");
```

(note how I'm not wrapping `users` with `[]`; I don't see why we want this array of user messages to be nested within another array, just for us to only have one element in this array at all times, if that makes sense)

File: betterdgg/modules/mentions.js [hunk 2]
```diff
@@ -7,9 +7,12 @@
...
+                        for (var i = 0; i < $(users)[0].length - 1; i++) {
+                            window.destiny.chat.gui.stylesheet.insertRule('.user-msg[data-username="' + ($($(users)[0].parent().parent()[i]).find(".user").text().toLowerCase()) + '"]{opacity:1 !important;}', window.destiny.chat.gui.stylesheet.cssRules.length);
```
Comment (kenany) [inline betterdgg/modules/mentions.js:12]: Does this stylesheet need to be removed when we click off of the user?

File: betterdgg/modules/mentions.js [hunk 3]
```diff
@@ -7,9 +7,12 @@
...
+                            window.destiny.chat.gui.focusedUsers.push(($($(users)[0].parent().parent()[i]).find(".user").text().toLowerCase()));
```
Comment (kenany) [inline betterdgg/modules/mentions.js:13]: I think I understand now that

``` js
($($(users)[0].parent().parent()[i]).find(".user").text().toLowerCase())
```

gives us the `i`th user to mention `username`. Could we make this its own variable in the loop, so that the lines aren't so long and also to help with readability?

File: betterdgg/modules/mentions.js (hunk 2 above)
Comment (kenany) [inline betterdgg/modules/mentions.js:12]: Furthermore, we have defined `gui = window.destiny.chat.gui` above so you should use that alias here.

**Review by kenany — CHANGES_REQUESTED** [review]
Only after staring at this for half an hour does it become even remotely clear what is going on here: just searching through all the chat lines for messages containing the selected username, and then applying an opacity of 1 to them. The use of `.parents()` and `.parent()` make this code very difficult to follow unless one actually checks out the DOM to see where each element is relative to each other. To be honest, I can't really suggest this be merged until it be made clearer as to what is going on here. I think my review comments would help a lot in that regard.

**Comment by PinkishCow** [comment]
``` javascript
var users = gui.lines.find(".chat-user:contains(" + username + ")"); 
```

This works in some cases but not others

Chat handles the clicking off itself because we use the same stuff they do

Using the predefined gui thing does not, I tested that when i first did it and tested it just now too.

Tried taking this out as a variable, did not work, not a clue why.

``` javascript
($($(users)[0].parent().parent()[i]).find(".user").text().toLowerCase())
```

Also gui.toggleUserFocus() already exists in the chat but it doesn't work no matter how hard you try with both the older version and the newer version.

**Comment by kenany** [comment]
Alright, I'll take another look at this Saturday or something. Might not be needed anyways in light of https://github.com/BryceMatthes/betterdgg/issues/55#issuecomment-247164036?

**Comment by PinkishCow** [comment]
Yeah it could also be the fault of it being 5am :P And yeah, that does fix the issue but personally i would like it to use the same method as the chat uses by default, as it does here, just incase.
